src/System.Collections/tests/Generic/HashSet/HashSet_ConstructorTests.cs

```diff
@@ -328,6 +328,108 @@ public static void Ctor_IEnumerable_HashSet_Multiple_Neg()
             driver.VerifyHashSet_NegativeTests();
         }
 
+        [Fact]
+        public static void Ctor_IEnumerable_HashSet_Fresh_HashSet()
+        {
+            HashSet<int> source = new HashSet<int>();
+            HashSet<int> hashSet = new HashSet<int>(source);
+
+            HashSetTestSupport<int> driver = new HashSetTestSupport<int>(hashSet, s_intGenerator, new int[0]);
+            driver.VerifyHashSetTests();
+        }
+
+        private static bool IsSquare(int value)
+        {
+            if (value == 0)
+                return true;
+            switch(value & 0x3F)
+            {
+                case 0:
+                case 1:
+                case 4:
+                case 9:
+                case 16:
+                case 17:
+                case 25:
+                case 33:
+                case 36:
+                case 41:
+                case 49:
+                case 57:
+                    int root = (int)Math.Sqrt(value);
+                    return value == root * root;
+                default:
+                    return false;
+            }
+        }
+
+        [Fact]
+        public static void Ctor_IEnumerable_HashSet_Sparse()
+        {
+            HashSet<int> source = new HashSet<int>();
+            for (int i = 0; i != 1000; ++i)
+            {
+                source.Add(i);
+            }
+            for (int i = 0; i != 1000; ++i)
+            {
+                if (!IsSquare(i)) // Unevenly spaced survivors increases chance of catching any spacing-related bugs.
+                {
+                    source.Remove(i);
+                }
+            }
+
+            HashSet<int> hashSet = new HashSet<int>(source);
+
+            HashSetTestSupport<int> driver = new HashSetTestSupport<int>(hashSet, s_intGenerator, new List<int>(source).ToArray());
+            driver.VerifyHashSetTests();
+        }
+
+        // Use parity only as a hashcode so as to have many collisions.
+        private class BadIntEqualityComparer : IEqualityComparer<int>
```

Comment:
Stylistically, I like nested types to be either at the top (for certain small things like enums) or at the bottom (usual).  I don't know that we really have a rule on it, though.  In the middle just feels weird.